### ISSUE TITLE
Implement Bors

### ISSUE THREAD

**Issue by matthiasbeyer** [thin-edge/thin-edge.io#1393]
Part of #1385 
Depends on #1391 
Depends on #1390 

---

This was briefly mentioned in [discussion 795](https://github.com/thin-edge/thin-edge.io/discussions/795), but never turned into something actionable.

So here we go: This is the tracking issue for implementing [bors](https://bors.tech).

---

Steps:

- [ ] Disable squash merges
- [ ] Consolidate all required CI actions in a "CI" workflow step (see tracking issue, see #1396 and #1395)
- [ ] Enable branch protection for "main"
    - [ ] With "Require a pull request before merging"
    - [ ] With "Require status checks to pass before merging"
    - [ ] With "bors" as only required status check
    - [ ] "Require pull request reviews before merging" MUST be unmarked
    - [ ] Required for administrators as well
- [ ] Adapt Codeowners setting to let bors handle Codeowners (related to #1391)

---

Note: Code owners MUST be implemented via bors:

> If you want to enforce reviews on your Pull Requests and/or you’re using CODEOWNERS, require these options solely on bors with the respective options: required_approvals and use_codeowners. Also, make sure bors is included in the list allowed to push to the protected branch.

**Comment by reubenmiller** [comment]
I just received a pop-up from GitHub informing about a their new-ish (currently public beta) "merge queue" feature. It sounds a lot like the functionality provided by bors.

https://github.blog/changelog/2021-10-27-pull-request-merge-queue-limited-beta/

**Comment by reubenmiller** [comment]
Closing in favour of using [Github merge queue](https://github.com/thin-edge/thin-edge.io/issues/2569)
